### DISCUSSION TITLE
wordpress__core-data

### DISCUSSION THREAD

**Discussion by tobeycodes** [DefinitelyTyped/DefinitelyTyped#67489] [Request a new @types package] [answered]
This package used to exist but was deprecated https://www.npmjs.com/package/@types/wordpress__core-data

The problem is that is their own type definitions are not sufficient. The other type packages for wordpress will definitely the data for actions and selectors https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/types/wordpress__blocks/index.d.ts#L9

```
declare module "@wordpress/data" {
    function dispatch(key: "core/blocks"): typeof import("./store/actions");
    function select(key: "core/blocks"): typeof import("./store/selectors");
}
```

This does not exist in the own type definitions so `select('core')` is undefined always.

**Comment by MysteryBlokHed** [comment, marked as answer]
Tooling will always prioritize builtin types over `@types` packages. If there's a problem with the typings, an issue should probably be raised at the original package's repository
